### PR TITLE
[frontend] Added new selectors

### PR DESCRIPTION
**Pull request by matiasmoreno876**
|Related issue|
|---|
|[1390](https://github.com/wazuh/wazuh-qa/issues/1390)|

## Description

Added new selectors for the wazuh menu.

### CODE REVIEW

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
+  decodersButton,
+  wazuhButton,
+  managementButton,
 } from '../../../pageobjects/wazuh-menu/wazuh-menu.page';
 
 When('The user navigates to decoders', () => {
-  getObject(wazuhButtonSelector)
-    .click();
-  getObject(menuListButtonsSelector)
-    .eq(0)
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js:13]: nice approach removing the ".eq(#)" and handling everything on the selector

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-    .click();
-  getObject(subMenuListButtonsSelector)
-    .eq(1)
-    .click();
-  cy.wait(3000);
+  clickElement(wazuhButton);
+  clickElement(managementButton);
+  clickElement(decodersButton);
+  cy.wait(5000);
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js:13]: we should find a way to stop waiting for an arbitrary amount of time.
Im working on it, but you can try it too if you like

https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/the-user-shou-id-see-the-message.then.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ import {
 } from '../../../pageobjects/wazuh-menu/decoders.page';
 
 Then('The user should see the message', () => {
+  cy.wait(5000);
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/the-user-shou-id-see-the-message.then.js:9]: at least add a TODO comment to remove them later when we implement a better logic

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+export const modulesButton = '.wz-menu-sections > :nth-child(1) > .euiButtonContent';
+export const managementButton = '.wz-menu-sections > :nth-child(2) > .euiButtonContent';
+export const agentsButton = '.wz-menu-sections > :nth-child(3) > .euiButtonContent';
+export const toolsButton = '.wz-menu-sections > :nth-child(4) > .euiButtonContent';
+export const securityButton = '.wz-menu-sections > :nth-child(6) > .euiButtonContent';
+export const settingsButton = '.wz-menu-sections > :nth-child(7) > .euiButtonContent';
+//endregion Menu
+
+//region SubMenu
+
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:13]: remove this extra line

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 5]
```diff
@@ -1,3 +1,74 @@
-export const wazuhButtonSelector = '.eui > .euiFlexGroup';
-export const menuListButtonsSelector = '[class="euiButtonEmpty euiButtonEmpty--text wz-menu-button "]';
-export const subMenuListButtonsSelector = '[class="euiSideNavItem euiSideNavItem--trunk"]';
+export const wazuhButton = '.eui > .euiFlexGroup';
+
+//region Menu
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:3]: good implementation of regions!
:D

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 6]
```diff
@@ -1,3 +1,74 @@
...
+export const modulesButton = '.wz-menu-sections > :nth-child(1) > .euiButtonContent';
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:4]: does this declarations follow any logic?
can we use alpha order? or should we follow the same order in which the elements are on the app?

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 7]
```diff
@@ -1,3 +1,74 @@
...
+
+//region Settings
+export const settingsApiConfigurationButton = '.euiSideNavItem__items > :nth-child(1) > .euiSideNavItemButton';
+export const settingsModulesButton = '.euiSideNavItem__items > :nth-child(2) > .euiSideNavItemButton';
+export const settingsSampleDataButton = '.euiSideNavItem__items > :nth-child(3) > .euiSideNavItemButton';
+export const settingsConfigurationButton = '.euiSideNavItem__items > :nth-child(4) > .euiSideNavItemButton';
+export const settingsLogsButton = '.euiSideNavItem__items > :nth-child(5) > .euiSideNavItemButton';
+export const settingsAboutButton = '.euiSideNavItem__items > :nth-child(6) > .euiSideNavItemButton';
+//endregion
+
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:73]: remove this extra line

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 8]
```diff
@@ -1,3 +1,74 @@
...
+//endregion
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:74]: add an empty line at the end of file

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 9]
```diff
@@ -1,3 +1,74 @@
-export const wazuhButtonSelector = '.eui > .euiFlexGroup';
-export const menuListButtonsSelector = '[class="euiButtonEmpty euiButtonEmpty--text wz-menu-button "]';
-export const subMenuListButtonsSelector = '[class="euiSideNavItem euiSideNavItem--trunk"]';
+export const wazuhButton = '.eui > .euiFlexGroup';
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:1]: this should be called wazuhMenuButton

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js [hunk 10]
```diff
@@ -1,3 +1,74 @@
...
+export const settingsButton = '.wz-menu-sections > :nth-child(7) > .euiButtonContent';
+//endregion Menu
+
+//region SubMenu
+
+//region Modules
+export const securityEventsButton = ':nth-child(1) > .euiSideNav > .euiSideNav__content > .euiSideNavItem--root > .euiSideNavItem__items > :nth-child(1) > .euiSideNavItemButton > .euiSideNavItemButton__content > .euiSideNavItemButton__label';
+export const integrityMonitoringButton = ':nth-child(1) > .euiSideNav > .euiSideNav__content > .euiSideNavItem--root > .euiSideNavItem__items > :nth-child(2) > .euiSideNavItemButton > .euiSideNavItemButton__content > .euiSideNavItemButton__label';
+export const amazonAwsButton = ':nth-child(1) > .euiSideNav > .euiSideNav__content > .euiSideNavItem--root > .euiSideNavItem__items > :nth-child(3) > .euiSideNavItemButton > .euiSideNavItemButton__content > .euiSideNavItemButton__label';
+export const googleCloudPlatformButton = ':nth-child(1) > .euiSideNav > .euiSideNav__content > .euiSideNavItem--root > .euiSideNavItem__items > :nth-child(4) > .euiSideNavItemButton > .euiSideNavItemButton__content > .euiSideNavItemButton__label';
```
Comment (MauGaP) [inline tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:18]: all these selectors will give us problems in the future... 
they are too attached to the page structure, like an absolut (ish) path

we need to modify the app to add data-cy attributes to these components.

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js (hunk 1 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js:13]: Thanks!!

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js (hunk 2 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/navigate-to-decoders.when.js:13]: Yes, that's my idea, completely eliminate the 'waiting'.

File: tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/the-user-shou-id-see-the-message.then.js (hunk 3 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/step-definitions/management/decoders/the-user-shou-id-see-the-message.then.js:9]: mmm, I don't think adding 'TODO' comments is a good approach, we should remove these functions when we find the best way to do it.

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js (hunk 6 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:4]: Currently follow the order of the app

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js (hunk 10 above)
Comment (frankeros) [reply tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:18]: if it deserves it, let's create an issue in wazuh-kibana-app

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js (hunk 9 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:1]: yes, good observation.

File: tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js (hunk 10 above)
Comment (matiasmoreno876) [reply tests/frontend/test_kibana_app/cypress/integration/pageobjects/wazuh-menu/wazuh-menu.page.js:18]: yes, exactly, we need to start adding more 'data-cy' attributes to be able to modify these selectors.

**Review by MauGaP — APPROVED** [review]

**Pull request merged**